src/discovery/Discovery.ts

```diff
@@ -112,7 +113,8 @@ class Discovery {
         // The sigchain data of the vertex (containing all cryptolinks)
         let vertexChainData: ChainData = {};
         // If the vertex we've found is our own node, we simply get our own chain
-        if (vertexGId.nodeId === this.nodeManager.getNodeId()) {
+        const nodeId = nodesUtils.decodeNodeId(vertexGId.nodeId)!;
```

Comment:
We make the assumption here that the `GestaltKey` stored in our local `GestaltGraph` will be containing a valid node ID. This seems fine.